src/codeu/chat/client/commandline/Chat.java

```diff
@@ -155,17 +176,17 @@ public void invoke(Scanner args) {
     //
     panel.register("u-sign-in", new Panel.Command() {
       @Override
-      public void invoke(Scanner args) {
-        final String name = args.hasNext() ? args.nextLine().trim() : "";
-        if (name.length() > 0) {
+      public void invoke(List<String> args) {
+        if(args.size() != 1){
+          System.out.println("ERROR: Failed to add new user");
+        }
+        else{
+          String name = args.get(0);
           final UserContext user = findUser(name);
           if (user == null) {
             System.out.format("ERROR: Failed to sign in as '%s'\n", name);
-          } else {
```

Comment:
We still need this else. Why you are removing it. We will do panels.push(createUserPanel(user)); only if the user is not null.